numpy/lib/arraysetops.py

```diff
@@ -518,21 +518,23 @@ def isin(elements, test_elements, **kwargs):
 
     Examples
     --------
-    >>> test = np.array([[0, 2], [4, 6]])
-    >>> states = [1, 2, 4, 8]
-    >>> mask = np.isin(test, states)
+    >>> elements = np.array([[0, 2], [4, 6]])
+    >>> test_elements = {1, 2, 4, 8}
+    >>> mask = np.isin(elements, test_elements)
     >>> mask
     array([[ False,  True],
            [ True,  False]], dtype=bool)
-    >>> test[mask]
+    >>> elements[mask]
     array([2, 4])
-    >>> mask = np.isin(test, states, invert=True)
+    >>> mask = np.isin(elements, test_elements, invert=True)
     >>> mask
     array([[ True, False],
            [ False, True]], dtype=bool)
-    >>> test[mask]
+    >>> elements[mask]
     array([0, 6])"""
     elements = np.array(elements)
+    if has_attr(test_elements, '__iter__') and not has_attr(test_elements, '__getitem__'):
+        test_elements = np.array([i for i in test_elements])
```

Comment:
I think we should this path for `dict` too, so that `'a' in dict(a=1)` matches `np.isin('a', dict(a=1))`. So maybe just `if not hasattr(test_elements, '__array__')`, to just ignore things that already know how to become arrays?